datadog_checks_dev/datadog_checks/dev/tooling/commands/meta/snmp/build_traps_db.py

```diff
@@ -0,0 +1,294 @@
+    the Datadog Agent to enrich trap data.
+    This command is intended for "Network Devices Monitoring" users who need to enrich traps that are not automatically
+    supported by Datadog.
+
+    The expected workflow is as such:\n
+    1- Identify a type of device that is sending traps that Datadog does not already recognize.\n
+    2- Fetch all the MIBs that we do not support.\n
+    3- Run `ddev meta snmp build-traps-db -o ./output_dir/ /path/to/my/mibs/*`\n
+    """
+
+    # Defaulting to github.com/DataDog/mibs.snmplabs.com/
+    mib_sources = [mib_sources] if mib_sources else [MIB_SOURCE_URL]
+
+    if output_file:
+        allowed_extensions = ['.json'] if use_json else ['.yml', '.yml']
+        if not any(output_file.endswith(x) for x in allowed_extensions):
+            echo_warning(
+                "Output file {} does not end with an allowed extension '{}'".format(
+                    output_file, ", ".join(allowed_extensions)
+                )
+            )
+            output_file = output_file + allowed_extensions[0]
+            echo_warning("Using {} instead.".format(output_file))
+
+    with ExitStack() as stack:
+
+        if not compiled_mibs_sources:
+            compiled_mibs_sources = stack.enter_context(TempDir('ddev_mibs'))
+
+        compiled_mibs_sources = os.path.abspath(compiled_mibs_sources)
+        if not os.path.isdir(compiled_mibs_sources):
+            abort("Output directory {} does not exist".format(compiled_mibs_sources))
+        echo_info("Writing intermediate compiled MIBs to {}".format(compiled_mibs_sources))
+
+        mibs_sources_dir = os.path.join(compiled_mibs_sources, 'mibs_sources')
+        if not os.path.isdir(mibs_sources_dir):
+            os.mkdir(mibs_sources_dir)
+
+        if output_dir and output_file:
+            abort("Do not set both --output-dir and --output-file at the same time.")
+        elif not output_file and not output_dir:
+            abort("Need to set one of --output-dir or --output-file")
+
+        mib_sources = (
+            sorted(set([os.path.abspath(os.path.dirname(x)) for x in mib_files if os.path.sep in x])) + mib_sources
+        )
+
+        mib_files = [os.path.basename(os.path.splitext(x)[0]) for x in mib_files if os.path.isfile(x)]
+        searchers = [AnyFileSearcher(compiled_mibs_sources).setOptions(exts=['.json'])]
+        code_generator = JsonCodeGen()
+        file_writer = FileWriter(compiled_mibs_sources).setOptions(suffix='.json')
+        mib_compiler = MibCompiler(SmiV1CompatParser(tempdir=''), code_generator, file_writer)
+        mib_compiler.addSources(*getReadersFromUrls(*mib_sources, **dict(fuzzyMatching=True)))
+        mib_compiler.addSearchers(*searchers)
+
+        try:
+            compiled_mibs, compiled_dependencies_mibs = compile_and_report_status(mib_files, mib_compiler)
+        except error.PySmiError:
+            abort("Unable to compile")
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Is it possible to know more about the error? it's location (at least which file)?

Reply:
Could help to add the debug option - something I found out while working on generator
https://github.com/DataDog/integrations-core/blob/master/datadog_checks_dev/datadog_checks/dev/tooling/commands/meta/snmp/generate_profile.py#L20
It lets mib compiler dump logs